### PR TITLE
Kubernetes detection support (wip: NOT TO BE MERGED)

### PR DESCRIPTION
**Pull request by felidadae**
Please do not merge this PR --> it will merged to master after allocation is merged.
---------------------------------------------------------------------------------------------------------
- [x] Detection
  - [x] pods discovery based on undocumented kubelet API 
  (https://github.com/intel/owca/pull/38#issuecomment-460178504)
  - [x] Implement KubernetesNode, KubernetesTask
  - [x] Implement ContainerSet class
  - [x] Test if working on a real kubernetes cluster @felidadae 
  - [x] Test if resctrl monitoring works @felidadae 
  - [x] ignore w get_tasks pods which are not fully ready (IMPORTANT) @mmucek95 
  - [x] get resources of the pod from kubelet @mmucek95 
  - [x] make KubernetesTask field name human friendly @mmucek95 
  - [x] fix previous unit tests
  - [x] add necessary new tests @mmucek95 and @felidadae 
    - [x] add test for get_tasks similar to test_mesos.py with fixture being a kubernetes response @mmucek95 
        - [x] corner case: not all containers are ready 
        - [x] corner case: not at all containers?? (pod with no containers?)
        - [x] test for find_pod_cgroup(self, pod_id, qos)
        - [x] test for find_resources
    - [x] add test for ContainerSet class @felidadae 
        - [x] check whether summing is done for perf events and cgroups
        - [x] check whether all pids of all sub containers are added to resgroup
    - [x] Add test to ContainerManager to check whether proper class Container or ContainerSet is created based on sub_cgroups field of Task @felidadae 
  - [x] fix all TODO
     - [x] find_pod_cgroup merge with find_container_cgroup @felidadae 
     - [x] add enum CGROUP_DRIVER__SYSTEMD CGROUP_DRIVER__CGROUPFS @mmucek95 

---------------------------------------------------------------------------------------------------------
Changes which should be removed from this PR to seperate ones:
- changes in owca/security.py
- remove file stress_ng deployment unless the new run workloads is not ready yet

### CODE REVIEW

File: owca/containers.py [hunk 1]
```diff
@@ -191,35 +349,46 @@ def sync_containers_state(self, tasks) -> Dict[Task, Container]:
...
-            container = Container(
-                new_task.cgroup_path,
-                rdt_enabled=self.rdt_enabled,
-                rdt_mb_control_enabled=self.rdt_mb_control_enabled,
-                platform_cpus=self.platform_cpus,
-                allocation_configuration=self.allocation_configuration,
-            )
+            # @TODO temporarily to make owca work
+            enable_multicontainer = True
+            if (enable_multicontainer and hasattr(new_task, "sub_cgroup_paths") and
```
Comment (ppalucki) [inline owca/containers.py]: task should have sub_cgroup_paths

File: owca/perf.py [hunk 2]
```diff
@@ -195,7 +195,7 @@ def _create_file_from_fd(pfd):
 class PerfCounters:
     """Perf facade on perf_event_open system call"""
 
-    def __init__(self, cgroup_path: str, event_names: List[MetricName]):
+    def __init__(self, cgroup_path: str, event_names: Iterable[MetricName]):
```
Comment (ppalucki) [inline owca/perf.py]: drop it

File: setup.py [hunk 3]
```diff
@@ -33,7 +33,8 @@
           'colorlog==3.1.4',
           'logging-tree==1.7',
           'dataclasses==0.6',
-          'confluent-kafka==0.11.4'
+          'confluent-kafka==0.11.4',
+          'kubernetes==8.0.1'
```
Comment (ppalucki) [inline setup.py]: drop

File: owca/containers.py [hunk 4]
```diff
@@ -53,75 +53,261 @@ def _convert_cgroup_path_to_resgroup_name(cgroup_path):
...
+        self._name = _convert_cgroup_path_to_resgroup_name(self._cgroup_path)
+        self._allocation_configuration = allocation_configuration
+        self._rdt_enabled = rdt_enabled
+        self._rdt_mb_control_enabled = rdt_mb_control_enabled
+        self._resgroup = resgroup
+        self._if_manage_resgroup = if_manage_resgroup
+
+        # Assert that implication is true:
+        #   self.if_manage_resgroup ==> self.resgroup is not None
+        #assert (not self._if_manage_resgroup or self._resgroup is not None)
```
Comment (felidadae) [inline owca/containers.py]: uncomment or remove

File: owca/containers.py [hunk 5]
```diff
@@ -191,35 +349,46 @@ def sync_containers_state(self, tasks) -> Dict[Task, Container]:
...
         for new_task in new_tasks:
-            container = Container(
-                new_task.cgroup_path,
-                rdt_enabled=self.rdt_enabled,
-                rdt_mb_control_enabled=self.rdt_mb_control_enabled,
-                platform_cpus=self.platform_cpus,
-                allocation_configuration=self.allocation_configuration,
-            )
+            # @TODO temporarily to make owca work
+            enable_multicontainer = True
```
[the remark this answers is not in the document]
Comment (felidadae) [reply owca/containers.py]: should be dropped

File: owca/containers.py (hunk 1 above)
Comment (felidadae) [reply owca/containers.py]: will be fixed by adding field to Task class
